templates/cluster-template-clusterclass.yaml

```diff
@@ -0,0 +1,280 @@
+apiVersion: cluster.x-k8s.io/v1beta1
```

Comment:
Consider using kustomize to generate the template and reuse this base flavor for the prow test template as a lot of it is shared

Reply:
@CecileRobertMichon Is there a way to patch a list using kustomize? I tried applying a patch to a list, but kustomize ends up replacing the entire list instead of merging it. [Example](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3166/commits/b3e516932f15c7affe90c15e3acc40bd203c4790#diff-c58e7ac2fdd1ff7852627da5e3b5e4cae9262265e522e4834aa7d978124427efL47)

Reply:
not with patchesStrategicMerge :/ 

you'll need to "patch" with the entire desired list

Reply:
or you can use `patchesJson6902`

https://stackoverflow.com/questions/71622419/adding-items-to-a-list-with-kubectl-kustomize